src/OSPSuite.Infrastructure.Import/Core/DataSource.cs

```diff
@@ -213,19 +213,20 @@ private void validateUnitsSupportedAndSameDimension(IReadOnlyList<ColumnInfo> co
                   //if unit comes from a column
                   if (column.Key.Column.Dimension == null)
                   {
-                     var dimensionOfFirstUnit = columnInfo.SupportedDimensions.FirstOrDefault(x => x.HasUnit(column.Value.ElementAt(0).Unit));
+                     var firstNonEmptyUnit = column.Value.FirstOrDefault(x => !string.IsNullOrEmpty(x.Unit));
+                     var dimensionOfFirstUnit = columnInfo.SupportedDimensions.FirstOrDefault(x => x.FindUnit(firstNonEmptyUnit.Unit, true) != null);
 
                      for (var i = 0; i < column.Value.Count(); i++)
                      {
                         if (double.IsNaN(column.Value.ElementAt(i).Measurement))
                            continue;
 
                         //if the unit specified does not belong to one of the supported dimensions of the mapping
-                        if (!columnInfo.SupportedDimensions.Any(x => x.HasUnit(column.Value.ElementAt(i).Unit)))
+                        if (!columnInfo.SupportedDimensions.Any(x => x.FindUnit(column.Value.ElementAt(i).Unit, true) != null))
```

Comment:
Since you are checkign if there is one and then retrieving it, what about you do something like this
var dimension = columnInfo.SupportedDimensions.FirstOrDefault(x => x.FindUnit(column.Value.ElementAt(i).Unit, ignoreCase: true)
if(dimension==null)
....
and later
if(dimension!=dimensionOfFirstUnit)

also column.Value.ElementAt(i) could be defined as a variable, for readability